services/etcd.go

```diff
@@ -242,14 +242,19 @@ func RemoveEtcdMember(ctx context.Context, toDeleteEtcdHost *hosts.Host, etcdHos
 			}
 			for _, member := range members.Members {
 				if member.Name == fmt.Sprintf("etcd-%s", toDeleteEtcdHost.HostnameOverride) {
-					mIDv3 = member.ID
+					mIDv3 = &member.ID
 					break
 				}
 			}
-			if _, err := etcdClient.MemberRemove(ctx, mIDv3); err != nil {
-				logrus.Debugf("Failed to list etcd members from host [%s]: %v", host.Address, err)
+			if mIDv3 == nil {
```

Comment:
`mIDv3` is declared outside the for loop, if the 1st interaction ( on the 1st host) fails but the etcd node is removed either by an external tool or the failure is due to a network delay, then when the 2nd interaction runs it will not find the member but the `mIDv3` is not nill so it invokes `MemberRemove` which will fail. Will we run into the same issue as the PR tries to fix?  We can avoid the case by declaring the variable inside the for loop.  The same to `mIDv2`.

Reply:
Refactored it a bit so this won't be an issue.